sphinx/ext/autosummary/generate.py

```diff
@@ -79,6 +80,11 @@ def _simple_info(msg):
 def _simple_warn(msg):
     print('WARNING: ' + msg, file=sys.stderr)
 
+def _underline(title, line='='):
+    if '\n' in title:
+        raise ValueError('Can only underline single lines')
+    return title + '\n' + line*len(title)
```

Comment:
But we don't use that in `apidoc` either. Than sounds like something for a separate PR, that also adds tests for these things.

Note that we already just use `len` in the `underscore` variable, so doing the same here isn't a regression.